packages/headless/src/api/generated-answer/generated-answer-client.ts

```diff
@@ -0,0 +1,152 @@
+import {EventSourcePolyfill} from 'event-source-polyfill';
+import {Logger} from 'pino';
+import {SearchAppState} from '../..';
+import {AsyncThunkOptions} from '../../app/async-thunk-options';
+import {ClientThunkExtraArguments} from '../../app/thunk-extra-arguments';
+import {SSEErrorPayload} from '../../features/generated-answer/generated-answer-actions';
+import {URLPath} from '../../utils/url-utils';
+import {resetTimeout} from '../../utils/utils';
+import {SearchAPIClient} from '../search/search-api-client';
+import {
+  GeneratedAnswerCitationsPayload,
+  GeneratedAnswerMessagePayload,
+  GeneratedAnswerPayloadType,
+  GeneratedAnswerStreamEventData,
+  GeneratedAnswerStreamFinishReason,
+} from './generated-answer-event-payload';
+import {GeneratedAnswerStreamRequest} from './generated-answer-request';
+
+export interface GeneratedAnswerAPIClientOptions {
+  logger: Logger;
+}
+
+export interface AsyncThunkGeneratedAnswerOptions<
+  T extends Partial<SearchAppState>
+> extends AsyncThunkOptions<
+    T,
+    ClientThunkExtraArguments<SearchAPIClient, GeneratedAnswerAPIClient>
+  > {}
+
+const buildStreamingUrl = (url: string, orgId: string, streamId: string) =>
+  new URLPath(
+    `${url}/rest/organizations/${orgId}/machinelearning/streaming/${streamId}`
```

Comment:
Is that service built with the mindset that it will be used for "everything" ML streaming related ? 🤔 

My question is mostly around naming of this thing (`GeneratedAnswerAPIClient`) and the path ( generic `machinelearning/streaming` which does not really contain anything related to "generated answer for search pages").

Reply:
it could be used for other types of streaming in the future but it is unlikely that the specs would be maintained so pretty much everything in this client is specific to "generated answer for search pages".